akka-http-core/src/main/scala/akka/http/impl/util/Java6Compat.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.impl.util
+
+import java.lang.reflect.{ InvocationTargetException, Method }
+import javax.net.ssl.SSLParameters
+
+import scala.util.control.NonFatal
+
+/**
+ * INTERNAL API
+ *
+ * Enables accessing SslParameters even if compiled against Java 6.
+ */
+private[http] object Java6Compat {
```

Comment:
:+1: There are definetly some more Candidates to be moved here. Would be nice to have a central place for them so we can get rid of the legacy stuff easier.